Statistics.hs

```diff
@@ -0,0 +1,224 @@
+    Just ba -> GH.github ba GH.issuesForRepoR githubOrg githubRepo GH.stateAll GH.FetchAll
+  issues <- case response of
+    Left err -> error $ show err
+    Right is -> pure is
+  case mCacheFile of
+    Nothing -> pure ()
+    Just cacheFile -> BL.writeFile cacheFile (encode issues)
+  pure issues
+
+data Issue = Issue
+  { issNumber :: !Int
+  , issTitle :: !Text
+  , issCreatedAt :: !UTCTime
+  , issClosedAt :: !(Maybe UTCTime)
+  , issLabels :: ![Text]
+  , issComments :: !Int
+  }
+  deriving (Eq, Ord, Show)
+
+githubIssueToIssue :: GH.Issue -> Issue
+githubIssueToIssue GH.Issue {..} = Issue {..}
+  where
+    issNumber = GH.unIssueNumber issueNumber
+    issTitle = issueTitle
+    issCreatedAt = issueCreatedAt
+    issClosedAt = issueClosedAt
+    issLabels = map (GH.untagName . GH.labelName) $ V.toList issueLabels
+    issComments = issueComments
+
+computeLifeTimeInDays :: Issue -> Maybe Double
+computeLifeTimeInDays Issue {..} = case issClosedAt of
+  Nothing -> Nothing
+  Just t -> Just $ realToFrac (diffUTCTime t issCreatedAt) / 86400
+
+computeDaysSinceCreation :: UTCTime -> Issue -> Double
+computeDaysSinceCreation currTime Issue {..} =
+  realToFrac (diffUTCTime currTime issCreatedAt) / 86400
+
+isApproved :: Issue -> Bool
+isApproved Issue {..} = "approved" `elem` issLabels
+
+isDeclined :: Issue -> Bool
+isDeclined Issue {..} = "declined" `elem` issLabels
+
+isMeta :: Issue -> Bool
+isMeta Issue {..} = "meta" `elem` issLabels
+
+isBase :: Int -> Issue -> Bool
+isBase n Issue {..} = ("base-4." <> T.pack (show n)) `elem` issLabels
+
+data Stat = Stat
+  { statMinIssue :: Issue
+  , statMinMetric :: Double
+  , statMinIssue2 :: Issue
+  , statMinMetric2 :: Double
+  , statMed :: Int
+  , statAvg :: Int
+  , statMaxIssue2 :: Issue
+  , statMaxMetric2 :: Double
+  , statMaxIssue :: Issue
+  , statMaxMetric :: Double
+  }
+  deriving (Show)
+
+collectStat :: (Issue -> Maybe Double) -> [Issue] -> Stat
+collectStat f is = Stat {..}
+  where
+    (statMinIssue, statMinMetric) = getMin is
+    (statMinIssue2, statMinMetric2) = getMin $ filter (/= statMinIssue) is
+
+    statMed = round $ median $ mapMaybe f is
+    statAvg = round $ average $ mapMaybe f is
+
+    (statMaxIssue, statMaxMetric) = getMax is
+    (statMaxIssue2, statMaxMetric2) = getMax $ filter (/= statMaxIssue) is
+
+    getExtremum g = swap . g . mapMaybe (\x -> (,x) <$> f x)
+    getMin = getExtremum minimum
+    getMax = getExtremum maximum
+
+    median xs = sort xs !! (length xs `quot` 2)
+    average xs = sum xs / fromIntegral (length xs)
+
+main :: IO ()
+main = do
+  cnf <-
+    execParser $
+      info
+        (configParser <**> helper)
+        (fullDesc <> header "Collect statistics for CLC proposals")
+  issues <- getGithubIssues cnf
+
+  let proposals =
+        filter (not . isMeta) . map githubIssueToIssue $
+          filter (isNothing . GH.issuePullRequest) $
+            V.toList issues
```

Comment:
I think this would include non-proposal issues like this: https://github.com/haskell/core-libraries-committee/issues/189

It does not carry the `meta` tag and I'm not sure it would be appropriate.